### PR TITLE
py3: more fixes for dicts and map

### PR DESCRIPTION
**Pull request by fchapoton**
some further fixe for dict.keys(), dict.values(), map()

### CODE REVIEW

**Comment by jwj61** [comment]
The change to web_number_field does not look right (just a typo).  It has fc and cf as dummy variables, and they should be the same.

File: lmfdb/backend/database.py [hunk 1]
```diff
@@ -797,7 +797,7 @@ def _check_header_lines(self, F, table_name, columns_set, sep=u"|", prohibit_mis
         col_list, col_type, _ = self._column_types(table_name)
         columns_set.discard("id")
         if not (columns_set <= set(col_list)):
-            raise ValueError("{} is not a subset of {}".format(columns_set, col_type.keys()))
```
Comment (edgarcosta) [inline lmfdb/backend/database.py]: We might give the wrong message here. (Nothing to do with the PR)

File: lmfdb/backend/database.py [hunk 2]
```diff
@@ -3397,7 +3397,7 @@ def upsert(self, query, data, commit=True):
                 for table, dat in cases:
                     inserter = SQL("INSERT INTO {0} ({1}) VALUES ({2})").format(
                                     Identifier(table),
-                                    SQL(", ").join(map(Identifier, dat.keys())),
+                                    SQL(", ").join(map(Identifier, list(dat.keys))),
```
Comment (edgarcosta) [inline lmfdb/backend/database.py]: `list(dat.keys)`-> `list(dat)`, and remove map

Comment (edgarcosta) [reply lmfdb/backend/database.py]: ~and remove map~

**Comment by edgarcosta** [comment]
Once tests pass I will merge it

**Pull request merged**